src/V3Cast.cpp

```diff
@@ -70,15 +70,19 @@ class CastVisitor final : public VNVisitor {
         nodep->user1(1);  // Now must be of known size
     }
     static int castSize(AstNode* nodep) {
-        if (nodep->isQuad()) {
-            return VL_QUADSIZE;
-        } else if (nodep->width() <= 8) {
-            return 8;
-        } else if (nodep->width() <= 16) {
-            return 16;
-        } else {
-            return VL_IDATASIZE;
-        }
+        if (nodep->isQuad()) return VL_QUADSIZE;
+        if (nodep->width() <= VL_BYTESIZE) return VL_BYTESIZE;
+        if (nodep->width() <= VL_SHORTSIZE) return VL_SHORTSIZE;
+        if (nodep->isWide()) return nodep->widthWords() * VL_EDATASIZE;
```

Comment:
A cast shouldn't care about the words involved, but maybe I'm confused.

Perhaps this should be a pre-pull?